comm/src/main/scala/coop/rchain/comm/network.scala

```diff
@@ -49,36 +49,41 @@ case class UnicastNetwork(id: NodeIdentifier, endpoint: Endpoint) extends Protoc
       table.observe(new ProtocolNode(sender, this))
       msg match {
         case ping @ PingMessage(_, _)     => handlePing(sender, ping)
-        case pong @ PongMessage(_, _)     => handlePong(sender, pong)
         case lookup @ LookupMessage(_, _) => handleLookup(sender, lookup)
-        case lookupResponse @ LookupResponseMessage(_, _) =>
-          handleLookupResponse(sender, lookupResponse)
+        case resp: ProtocolResponse => handleResponse(sender, resp)
       }
     }
 
   /**
-    * Validate incoming PING and send responding PONG.
+    *
     */
-  private def handlePing(sender: PeerNode, ping: PingMessage): Unit =
+  private def handleResponse(sender: PeerNode, msg: ProtocolResponse): Unit =
     for {
-      pong <- ping.response(local)
+      ret <- msg.returnHeader
```

Comment:
This is using the Option monad, which means if there's no return header or pending.get returns None, this does nothing.  When would this function be called in a situation where msg.returnHeader is None?  Would that be an error condition?

Reply:
It might be helpful to log the occurrence during development, but in general I expect traffic to be quite dirty, since we won't have control over the sender. I have been mulling a verbose-flag for that.